src/loaders/fixImageLoader.js

```diff
@@ -12,14 +15,44 @@
  *
  * @type {(this: import("webpack").LoaderContext<{}>, content: string) => string>}
  */
-function fixImageLoader(content) {
-  // replace(/(.+\:\/\/[^\/]+){0,1}\/.*/i, '$1')
-  //    this regexp will extract the hostname from publicPath
-  //    http://localhost:3000/_next/... -> http://localhost:3000
-  const currentHostnameCode =
-    "__webpack_require__.p.replace(/(.+\\:\\/\\/[^\\/]+){0,1}\\/.*/i, '$1')";
+async function fixImageLoader(remaining) {
+  this.cacheable(true);
+  const isServer = this._compiler.options.name !== 'client';
+  const result = await this.importModule(
+    `${this.resourcePath}.webpack[javascript/auto]` + `!=!${remaining}`
+  );
+  const content = result.default || result;
+
+  const computedAssetPrefix = isServer
+    ? `(global.__dynamicAssetPrefix && global.__dynamicAssetPrefix.indexOf(\'://\') > 0 ? new URL(global.__dynamicAssetPrefix).origin : \'\')`
```

Comment:
Global asset prefix is lululemon specific. Let's not use that line on this loader. Just manage client side conditions for now

Reply:
We can hook into my SSR equivalent for public path auto once we start working on SSR. Then that can be integrated in place of dynamic asset prefix global

Reply:
great point, changed accordingly